### DISCUSSION TITLE
NVreg_EnableGpuFirmware=0 not work in open nvidia driver?

### DISCUSSION THREAD

**Discussion by vxuyu** [NVIDIA/open-gpu-kernel-modules#667] [Q&A] [answered]
Insmod with NVreg_EnableGpuFirmware=0, and /proc/driver/nvidia/params shows "EnableGpuFirmware: 0", while `nvidia-smi --query-gpu=gsp.mode.current --format=csv` still shows Enabled.

Is it on purpose? Where can I find relevant documentation?

**Comment by mtijanic** [comment, marked as answer]
> NVreg_EnableGpuFirmware=0 not work in open nvidia driver?

Correct. Open kernel modules _require_ the use of GSP firmware and can't work without it. Maybe in a future release we can add a more visible warning if you try to set that parameter.

**Reply by vxuyu** [reply]
Thanks a lot!

**Reply by TopchetoEU** [reply]
it would be nice if a piece of software that feels to me is still in early alpha, that was previously opt-out, didn't become a backbone part of the driver, but it's nvidia after all, what can you do about it. they certainly don't have any resources/talented engineers to work out the kinks of the linux driver.

**Comment by hrushirajg23** [comment]
does anyone know how to solve this issue

[ 10.842101] NVRM: s_vbiosPatchInterfaceData: Found pIntFaceHdr entry count: 0, expected 2
[ 10.842107] NVRM: s_vbiosPatchInterfaceData: too few interface entires found for FWSEC cmd 0x15
[ 10.842112] NVRM: s_prepareForFwsec_TU102: Falcon ucode from hs
[ 10.842114] NVRM: s_prepareForFwsec_TU102: failed to prepare interface data for FWSEC cmd 0x15: 0x25
[ 10.842116] NVRM: s_prepareForFwsec_TU102: (note: VBIOS version 94.02.71.40.83)
[ 10.842119] NVRM: nvCheckOkFailedNoLog: Check failed: Invalid data passed [NV_ERR_INVALID_DATA] (0x00000025) returned from kgspPrepareForBootstrap_HAL(pGpu, pKernelGsp, KGSP_BOOT_MODE_NORMAL) @ kernel_gsp.c:3664
[ 10.842171] NVRM: RmInitAdapter: Cannot initialize GSP firmware RM
[ 10.843941] NVRM: GPU 0000:01:00.0: RmInitAdapter failed! (0x62:0x25:2015)
[ 10.845693] NVRM: GPU 0000:01:00.0: rm_init_adapter failed, device minor number 0

Can somebody at-least explain what is the reason behind this "firmware initialization issue" ?
I'm trying to run rtx 3080 on rk3588 sbc .
I added my chipset info to fix the following issue:
[ 6.935721] NVRM: loading NVIDIA UNIX aarch64 Kernel Module 580.95.05 Tue Sep 23 10:16:20 UTC 2025
[ 6.956390] NVRM: Chipset not recognized (vendor ID 0x1d87, device ID 0x3588)
[ 6.956397] The NVIDIA GPU driver for AArch64 has not been qualified on this platform
and therefore it is not recommended or intended for use in any production
environment.

while running proprietary driver I get only:

the chipset error
and
[ 71.348099] NVRM: GPU 0000:01:00.0: RmInitAdapter failed! (0x25:0x65:1623)
[ 71.348111] NVRM: GPU 0000:01:00.0: rm_init_adapter failed, device minor number 0
[ 71.366226] NVRM: GPU at PCI:0000:01:00: GPU-7963b590-e5ce-1751-e3f8-c6d771836da2
[ 71.366233] NVRM: Xid (PCI:0000:01:00): 79, GPU has fallen off the bus.
[ 71.366237] NVRM: GPU 0000:01:00.0: GPU has fallen off the bus.
[ 71.366269] NVRM: A GPU crash dump has been created. If possible, please run
NVRM: nvidia-bug-report.sh as root to collect this data before
NVRM: the NVIDIA kernel module is unloaded.

Thanks
